app/src/main/java/protect/card_locker/ManageGroupActivity.java

```diff
@@ -0,0 +1,264 @@
+    private String mCurrentGroupName;
+
+    private boolean mGroupNameNotInUse;
+
+    @Override
+    protected void onCreate(Bundle inputSavedInstanceState)
+    {
+        super.onCreate(inputSavedInstanceState);
+        setContentView(R.layout.activity_manage_group);
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        mHelpText = findViewById(R.id.helpText);
+        mCardList = findViewById(R.id.list);
+        FloatingActionButton saveButton = findViewById(R.id.fabSave);
+
+        mGroupNameText = findViewById(R.id.editTextGroupName);
+
+        Intent intent = getIntent();
+        String groupId = intent.getStringExtra("group");
+        if (groupId == null){
+            throw(new IllegalArgumentException("this activity expects a group loaded into it's intent"));
+        }
+        Log.d("groupId", "groupId: " + groupId);
+        mGroup = mDB.getGroup(groupId);
+        if (mGroup == null){
+            throw(new IllegalArgumentException("cannot load group " + groupId + " from database"));
+        }
+        mAdapter = new ManageGroupCursorAdapter(this, null, this, mGroup);
+        mCardList.setAdapter(mAdapter);
+        registerForContextMenu(mCardList);
+
+        if (inputSavedInstanceState != null) {
+            ArrayList<Integer> adapterState = inputSavedInstanceState.getIntegerArrayList("mAdapterState");
+            if (adapterState != null) {
+                integerArrayToAdapterState(adapterState);
+            }
+            mCurrentGroupName = inputSavedInstanceState.getString("mCurrentGroupName");
+        }
+
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar == null){
+            throw(new RuntimeException("mActionBar is not expected to be null here"));
+        }
+        actionBar.setDisplayHomeAsUpEnabled(true);
+        actionBar.setDisplayShowHomeEnabled(true);
+
+        saveButton.setOnClickListener(v -> {
+            String currentGroupName = mGroupNameText.getText().toString();
+            if(!currentGroupName.trim().equals(mGroup._id)){
+                if(!mGroupNameNotInUse) {
+                    Toast toast = Toast.makeText(getApplicationContext(), R.string.group_name_already_in_use, Toast.LENGTH_SHORT);
+                    toast.show();
+                    return;
+                }
+                if(currentGroupName.trim().length() == 0){
+                    Toast toast = Toast.makeText(getApplicationContext(), R.string.group_name_is_empty, Toast.LENGTH_SHORT);
+                    toast.show();
+                    return;
+                }
+            }
+
+            mAdapter.commitToDatabase();
+            Toast toast = Toast.makeText(getApplicationContext(), R.string.group_updated, Toast.LENGTH_SHORT);
+            if(!currentGroupName.trim().equals(mGroup._id)){
+                mDB.updateGroup(mGroup._id, currentGroupName.trim());
+            }
+            toast.show();
```

Comment:
This can just be a single line, running .show() right after the makeText. No need to save it to a temporary variable.